architecture/core.rst

```diff
@@ -390,6 +390,149 @@ normal world supports scheduling of processes. Even on UP systems, supporting
 several trusted threads in optee_os helps normal world scheduler to be
 efficient.
 
+Core handlers for native interrupts
+===================================
+
+OP-TEE core provides methods for device drivers to setup and register
+handler functions for native interrupt controller drivers
+(see:ref:`native_foreign_irqs`).
+Interrupt handlers can be nested for when an interrupt controller
```

Comment:
The `for` probably shouldn't be there, what about changing 399-401  to:
_Interrupt handlers can be nested when an interrupt controller exposes interrupts, which are multiplexed on an interrupt controlled by a parent interrupt controller._